config/credhub_secret_test.go

```diff
@@ -0,0 +1,147 @@
+package config_test
+
+import (
+	"code.cloudfoundry.org/cli/plugin/pluginfakes"
+	"errors"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pivotal-cf/spring-cloud-services-cli-plugin/config"
+	"github.com/pivotal-cf/spring-cloud-services-cli-plugin/httpclient/httpclientfakes"
+	"github.com/pivotal-cf/spring-cloud-services-cli-plugin/serviceutil/serviceutilfakes"
+)
+
+var _ = Describe("CredhubSecret", func() {
+
+	const (
+		accessToken       = "fake-access-token"
+		bearerAccessToken = "bearer fake-access-token"
+		serviceURI        = "service-uri/"
+		secretsURI        = "service-uri/secrets"
+		configServerName  = "fake-config-server-name"
+	)
+
+	var (
+		fakeCliConnection *pluginfakes.FakeCliConnection
+		credhubSecret     config.CredHubSecret
+		fakeAuthClient    *httpclientfakes.FakeAuthenticatedClient
+		fakeResolver      *serviceutilfakes.FakeServiceInstanceResolver
+		secretsError      error
+	)
+
+	BeforeEach(func() {
+		fakeCliConnection = &pluginfakes.FakeCliConnection{}
+		fakeAuthClient = &httpclientfakes.FakeAuthenticatedClient{}
+		fakeResolver = &serviceutilfakes.FakeServiceInstanceResolver{}
+
+		fakeCliConnection.AccessTokenReturns(bearerAccessToken, nil)
+		fakeResolver.GetServiceInstanceUrlReturns(serviceURI, nil)
+	})
+
+	JustBeforeEach(func() {
+		credhubSecret = config.NewCredHubSecret(fakeCliConnection, fakeAuthClient, fakeResolver)
+	})
+
+	Describe("CredHub Add Secret", func() {
+
+		JustBeforeEach(func() {
+			secretsError = credhubSecret.Add(configServerName, "application/cloud/one", "{\"key\":\"secret\"}")
+		})
+
+		Context("when add returns success", func() {
```

Comment:
Tempted to simplify a bit and take the happy path test out of a context, just to make this read a bit better. 'When add returns success it should call secrets endpoint' doesn't really make sense. 'It calls the secrets endpoint' should be enough I think?